service/src/main/java/com/codeforcommunity/requester/MapRequester.java

```diff
@@ -41,85 +46,132 @@ public MapRequester(Vertx vertx) {
+                      .add("outFields", "FID");
 
               return Future.future(
                   promise -> {
                     client
-                        .postAbs(featureLayerRoute)
+                        .postAbs(featureLayerQueryRoute)
                         .putHeader("content-type", "multipart/form-data")
                         .sendForm(
                             formData,
-                            (ar) -> {
-                              if (ar.succeeded()) {
-                                HttpResponse<Buffer> httpResponse = ar.result();
+                            (asyncResult -> {
+                              if (asyncResult.succeeded()) {
+                                HttpResponse<Buffer> httpResponse = asyncResult.result();
 
                                 if (httpResponse.statusCode() == 200) {
                                   JsonObject responseBody = httpResponse.bodyAsJsonObject();
-                                  if (responseBody.containsKey("error")
-                                      && responseBody.getJsonObject("error").getInteger("code")
-                                          == 498) {
-                                    // The API token is invalid, reset it and make this call again.
-                                    logger.info(
-                                        "Remaking token request while updating blocks to "
-                                            + updateTo.name());
-                                    this.tokenFuture = updateToken();
-                                    updateLayers(streetFids, updateTo, this.tokenFuture)
-                                        .onSuccess((V) -> promise.complete())
-                                        .onFailure(promise::fail);
-                                  } else if (responseBody.containsKey("updateResults")) {
-                                    logger.info(
-                                        "Successfully updated blocks to " + updateTo.name());
-                                    // Check successes
-                                    promise.complete();
-                                  } else {
-                                    // throw error
-                                    logger.error(
-                                        "Update street request responded with unrecognized response body: "
-                                            + responseBody);
+                                  if (responseBody.containsKey("features")) {
+                                    List<Map> features =
+                                        responseBody.getJsonArray("features").getList();
+                                    JsonArray updateJson = new JsonArray();
+                                    features.forEach(
+                                        feature -> {
+                                          Map attributes = (Map) feature.get("attributes");
```

Comment:
This get should also have an if statement before it in case we get some weird case where the attributes key is missing, same behavior should be followed as if the features key was missing, i.e. log the result and continue the function

Reply:
This if/else logic seems pretty suspect IMO. If an error is going to occur, it's going to occur, so we should let the exception throw and the handler will catch it. @justinkonecny let me know if I'm misinterpreting what you said about the logger regarding this file.